.github/workflows/test-workflow.yml

```diff
@@ -0,0 +1,81 @@
+on: 
+  push:
+    branches:
+      - '**'
+    tags-ignore:
+      - '*.*'
+  pull_request:
+    branches:
+      - '**'
+env:
+  factorio_password: ${{ secrets.FACTORIO_PASSWORD }}
```

Comment:
Can probably just make these generic strings instead of secrets for the tests.

Reply:
When you want your factorio credentials in a public githu repo, sure, do it. But i want to keep my username and password a secret :)

Reply:
Ah didn't realize valid credentials were needed for the tests.